### ISSUE TITLE
Python throws PTX linker error when using `print` after autodiff for Float64

### ISSUE THREAD

**Issue by realWDC** [mitsuba-renderer/enoki#108]
Following code works in `Float32`:
```Python
import enoki as ek
from enoki.cuda_autodiff import Float32 as FloatD # works
a = FloatD(2.1234567890)
print(a) # works
ek.set_requires_gradient(a)
b = a * a
ek.backward(b)
print(b)
```
For some reason I need more precision, but the same code in `Float64` will throw an error:
```Python
import enoki as ek
from enoki.cuda_autodiff import Float64 as FloatD # not works when print
a = FloatD(2.1234567890)
print(a) # works but does not output all digits
ek.set_requires_gradient(a)
b = a * a
ek.backward(b)
print(b)              # error
```
which outputs in terminal:
```
[2.12346]
PTX linker error:
ptxas application ptx input, line 47; error   : Arguments mismatch for instruction 'setp'
ptxas application ptx input, line 51; error   : Arguments mismatch for instruction 'setp'
ptxas fatal   : Ptx assembly aborted due to errors
cuda_check(): driver API error = 0400 "CUDA_ERROR_INVALID_HANDLE" in /enoki/src/cuda/jit.cu:253.
```
Here is a C++ version of above two, and it runs perfectly for both cases:
```c++
#include <iostream>
#include <iomanip>
#include <enoki/autodiff.h>
using namespace enoki;
// using Float  = float;  // works
using Float  = double; // also works
using FloatD = DiffArray<Float>;
int main() {
    FloatD a(2.1234567890);
    std::cout << std::setprecision(17) << a << std::endl;
    set_requires_gradient(a);
    FloatD b = a * a;
    backward(b);
    std::cout << b << std::endl;
}
```
I think there is a small issue in Python when using `print()` for `Float64`. I am on Ubuntu 18.04 with TITAN X (Pascal), Cuda 10.2 with Enoki compiled with clang++-9.

**Comment by Speierers** [comment]
The difference between you Python and C++ code is that in the first one, you are working with `DiffArray<CUDAArray<double>>` while in the second one your are working with `DiffArray<double>`.

To my knowledge `CUDAArray<double>` aren't supported in enoki.

For single values, you could try to first convert it `FloatD -> float` and then print. Something like `print(b[0])`.

**Comment by wjakob** [comment]
`CUDAArray<double>` is not well-tested in this release of Enoki, and it causes issues together with Autodiff. We're currently refactoring Enoki and will release a new version soon that is going to fix this issue, amongst other things.

**Comment by realWDC** [comment]
Thank you for your replies.
After @Speierers's comment I realized I made an embarrassing mistake (missing `CUDAArray`) in the C++ example. Fixing which the C++ version yields similar error as in Python's version.

So my question makes no sense because Enoki does not support `double` in CUDA Autodiff. This is reasonable because `double` are not efficient on GPUs and also `double` is not well-supported by OptiX (in Mitsuba2).

Looking forward to the new Enoki and hopefullly `double` can be supported in Enoki Autodiff (... though not necessary for Mitsuba2).
